pkg/gitparse/gitparse.go

```diff
@@ -53,6 +131,51 @@ type Parser struct {
 	dateFormat    string
 }
 
+// noOpCloser wraps an io.Reader to add a no-op Close method, forming an io.ReadCloser.
+type noOpCloser struct{ io.Reader }
+
+// Close performs no operation (no-op) and returns nil.
+// It's used to fulfill the io.Closer interface.
+func (noc *noOpCloser) Close() error { return nil }
+
+// DiffContentReadCloser returns an io.ReadCloser for reading the content of a Diff.
+// If the diff content size exceeds a predefined threshold, it is stored in a temporary file,
+// and the function returns an auto-deleting file reader (newAutoDeletingFileReader) to read from this file.
+// For smaller diffs that fit within the threshold, the content is kept in memory,
+// and the function returns a no-op closer wrapper (noOpCloser) around a bytes.Reader.
+// The caller is responsible for calling Close on the returned io.ReadCloser in both cases.
+func DiffContentReadCloser(d *Diff) (io.ReadCloser, error) {
+	if d.streamDestination != nil {
+		// Data is in a file, read from the file.
+		file, err := os.Open(d.streamDestination.Name())
+		if err != nil {
+			return nil, err
+		}
+		return newAutoDeletingFileReader(file), nil
+	}
+	// Data is in memory.
+	return &noOpCloser{Reader: bytes.NewReader(d.Content.Bytes())}, nil
+}
```

Comment:
Can this function only be called after we finish parsing the diff (i.e. after `finalize()` is called)?

Reply:
yea that sounds accurate. 👍